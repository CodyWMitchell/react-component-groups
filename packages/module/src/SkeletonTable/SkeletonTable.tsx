import React, { ReactNode } from 'react';
import { Caption, Table, TableProps, TableVariant, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
import { Skeleton } from '@patternfly/react-core';

export type SkeletonTableProps = Omit<TableProps, 'ref'> & {
  /** Indicates the table variant */
  variant?: TableVariant;
  /** The number of rows the skeleton table should contain */
  rows?: number;
  /** Any captions that should be added to the table */
  caption?: ReactNode;
  /** Custom OUIA ID */
  ouiaId?: string | number;
  /** Indicates if the table has checkboxes */
  hasCheckbox?: boolean;
  /** Indicates if the table has expandable rows */
  isExpandable?: boolean;
  /** The index of the sorted column */
  sortIndex?: number;
} & (
    | {
        columns: {
          title: ReactNode;
          isSortable?: boolean;
        }[];
      }
    | {
        numberOfColumns: number;
      }
  );

function hasCustomColumns(props: Record<string, any>): props is SkeletonTableProps & {
  columns: {
    title: ReactNode;
    isSortable?: boolean;
  }[];
} {
  return Array.isArray(props.columns) && props.columns.some((column) => typeof column === 'object');
}

const SkeletonTable: React.FunctionComponent<SkeletonTableProps> = (props: SkeletonTableProps) => {
  const {
    variant,
    rows = 5,
    caption,
    ouiaId = 'SkeletonTable',
    hasCheckbox = false,
    isExpandable = false,
    sortIndex,
    ...rest
  } = props;
  const rowCells = hasCustomColumns(props) ? props.columns.length : props.numberOfColumns;
  const rowArray = [ ...new Array(rowCells) ];
  const bodyRows = [ ...new Array(rows) ].map((_, rowIndex) => (
    <Tr key={rowIndex} ouiaId={`${ouiaId}-tr-${rowIndex}`}>
      {hasCheckbox && (
        <Td key="checkbox" data-ouia-component-id={`${ouiaId}-td-${rowIndex}-checkbox`}>
          <Skeleton width="16px" />
        </Td>
      )}
      {isExpandable && (
        <Td key="expandable" data-ouia-component-id={`${ouiaId}-td-${rowIndex}-expandable`}>
          <Skeleton width="16px" />
        </Td>
      )}
      {rowArray.map((_, colIndex) => (
        <Td key={colIndex} data-ouia-component-id={`${ouiaId}-td-${rowIndex}-${colIndex}`}>
          <Skeleton />
        </Td>
      ))}
    </Tr>
  ));

  return (
    <Table aria-label="Loading" variant={variant} ouiaId={ouiaId} {...rest}>
      {caption && <Caption>{caption}</Caption>}
      <Thead data-ouia-component-id={`${ouiaId}-thead`}>
        <Tr ouiaId={`${ouiaId}-tr-head`}>
          {hasCheckbox && <Th key="checkbox" data-ouia-component-id={`${ouiaId}-th-checkbox`} />}
          {isExpandable && <Th key="expandable" data-ouia-component-id={`${ouiaId}-th-expandable`} />}
          {hasCustomColumns(props)
            ? props.columns.map((column, index) => (
              <Th
                key={index}
                data-ouia-component-id={`${ouiaId}-th-${index}`}
                className={index === sortIndex ? 'pf-m-selected' : ''}
              >
                {column.title}
                {column.isSortable && (
                  <span className="pf-c-table__sort-indicator">
                    <Skeleton width="16px" />
                  </span>
                )}
              </Th>
            ))
            : rowArray.map((_, index) => (
              <Th key={index} data-ouia-component-id={`${ouiaId}-th-${index}`}>
                <Skeleton />
              </Th>
            ))}
        </Tr>
      </Thead>
      <Tbody data-ouia-component-id={`${ouiaId}-tbody`}>{bodyRows}</Tbody>
    </Table>
  );
};

export default SkeletonTable;
