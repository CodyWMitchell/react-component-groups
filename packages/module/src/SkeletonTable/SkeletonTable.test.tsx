import React from 'react';
import { render } from '@testing-library/react';
import SkeletonTable from './SkeletonTable';

describe('SkeletonTable component', () => {
  it('should render correctly', () => {
    expect(render(<SkeletonTable columns={[ { title: 'first' }, { title: 'second' } ]} />)).toMatchSnapshot();
  });

  it('should render correctly with rows', () => {
    expect(render(<SkeletonTable columns={[ { title: 'first' }, { title: 'second' } ]} rows={5} />)).toMatchSnapshot();
  });

  it('should render correctly with a caption', () => {
    expect(
      render(<SkeletonTable columns={[ { title: 'first' }, { title: 'second' } ]} caption="Test Caption" />)
    ).toMatchSnapshot();
  });

  it('should render correctly with a custom OUAI ID', () => {
    expect(
      render(<SkeletonTable columns={[ { title: 'first' }, { title: 'second' } ]} ouiaId="customID" />)
    ).toMatchSnapshot();
  });

  it('should render correctly with checkboxes', () => {
    expect(render(<SkeletonTable columns={[ { title: 'first' }, { title: 'second' } ]} hasCheckbox />)).toMatchSnapshot();
  });

  it('should render correctly with expandable rows', () => {
    expect(
      render(<SkeletonTable columns={[ { title: 'first' }, { title: 'second' } ]} isExpandable />)
    ).toMatchSnapshot();
  });

  it('should render correctly with a sort index', () => {
    expect(
      render(<SkeletonTable columns={[ { title: 'first' }, { title: 'second' } ]} sortIndex={0} />)
    ).toMatchSnapshot();
  });
});
