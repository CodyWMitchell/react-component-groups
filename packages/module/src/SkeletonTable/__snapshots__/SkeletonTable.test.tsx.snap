// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonTable component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        aria-label="Loading"
        class="pf-v5-c-table pf-m-grid-md"
        data-ouia-component-id="OUIA-Generated-Table-1"
        data-ouia-component-type="PF5/Table"
        data-ouia-safe="true"
        role="grid"
      >
        <thead
          class="pf-v5-c-table__thead"
        >
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-1"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <th
              class="pf-v5-c-table__th"
              scope="col"
              tabindex="-1"
            >
              first
            </th>
            <th
              class="pf-v5-c-table__th"
              scope="col"
              tabindex="-1"
            >
              second
            </th>
          </tr>
        </thead>
        <tbody
          class="pf-v5-c-table__tbody"
          role="rowgroup"
        >
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-2"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-3"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-4"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-5"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-6"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      aria-label="Loading"
      class="pf-v5-c-table pf-m-grid-md"
      data-ouia-component-id="OUIA-Generated-Table-1"
      data-ouia-component-type="PF5/Table"
      data-ouia-safe="true"
      role="grid"
    >
      <thead
        class="pf-v5-c-table__thead"
      >
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-1"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <th
            class="pf-v5-c-table__th"
            scope="col"
            tabindex="-1"
          >
            first
          </th>
          <th
            class="pf-v5-c-table__th"
            scope="col"
            tabindex="-1"
          >
            second
          </th>
        </tr>
      </thead>
      <tbody
        class="pf-v5-c-table__tbody"
        role="rowgroup"
      >
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-2"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-3"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-4"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-5"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-6"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SkeletonTable component should render correctly with rows 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        aria-label="Loading"
        class="pf-v5-c-table pf-m-grid-md"
        data-ouia-component-id="OUIA-Generated-Table-2"
        data-ouia-component-type="PF5/Table"
        data-ouia-safe="true"
        role="grid"
      >
        <thead
          class="pf-v5-c-table__thead"
        >
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-7"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <th
              class="pf-v5-c-table__th"
              scope="col"
              tabindex="-1"
            >
              first
            </th>
            <th
              class="pf-v5-c-table__th"
              scope="col"
              tabindex="-1"
            >
              second
            </th>
          </tr>
        </thead>
        <tbody
          class="pf-v5-c-table__tbody"
          role="rowgroup"
        >
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-8"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-9"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-10"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-11"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
          <tr
            class="pf-v5-c-table__tr"
            data-ouia-component-id="OUIA-Generated-TableRow-12"
            data-ouia-component-type="PF5/TableRow"
            data-ouia-safe="true"
          >
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
            <td
              class="pf-v5-c-table__td"
              tabindex="-1"
            >
              <div
                class="pf-v5-c-skeleton"
              >
                <span
                  class="pf-v5-u-screen-reader"
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      aria-label="Loading"
      class="pf-v5-c-table pf-m-grid-md"
      data-ouia-component-id="OUIA-Generated-Table-2"
      data-ouia-component-type="PF5/Table"
      data-ouia-safe="true"
      role="grid"
    >
      <thead
        class="pf-v5-c-table__thead"
      >
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-7"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <th
            class="pf-v5-c-table__th"
            scope="col"
            tabindex="-1"
          >
            first
          </th>
          <th
            class="pf-v5-c-table__th"
            scope="col"
            tabindex="-1"
          >
            second
          </th>
        </tr>
      </thead>
      <tbody
        class="pf-v5-c-table__tbody"
        role="rowgroup"
      >
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-8"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-9"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-10"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-11"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
        <tr
          class="pf-v5-c-table__tr"
          data-ouia-component-id="OUIA-Generated-TableRow-12"
          data-ouia-component-type="PF5/TableRow"
          data-ouia-safe="true"
        >
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
          <td
            class="pf-v5-c-table__td"
            tabindex="-1"
          >
            <div
              class="pf-v5-c-skeleton"
            >
              <span
                class="pf-v5-u-screen-reader"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
